packages/actor-query-operation-quadpattern/test/ActorQueryOperationQuadpattern-test.ts

```diff
@@ -760,5 +809,152 @@ describe('ActorQueryOperationQuadpattern', () => {
         ],
       );
     });
+
+    it('should run <<s p o>> ?p o ?g', async() => {
+      mediatorResolveQuadPattern.mediate = jest.fn(
+        () => {
+          const data = new ArrayIterator([
+            quad('<<s p o>>', 'p1', 'o1', 'g1'),
+            quad('<<s p o>>', 'p2', 'o2', 'g2'),
+            quad('<<s p o>>', 'p3', 'o3', 'g3'),
+            quad('sd', 'pd', 'od'),
+          ]);
+          data.setProperty('metadata', metadataContent);
+          return Promise.resolve({ data });
+        },
+      );
+
+      const operation = AF.createPattern(
+        DF.quad(DF.namedNode('s'), DF.namedNode('p'), DF.namedNode('o')),
+        DF.variable('p'),
+        DF.namedNode('o'),
+        DF.variable('g'),
+      );
+      const output = <IQueryOperationResultBindings> await actor.run({ operation, context });
+      expect(await output.metadata()).toEqual({
+        cardinality: { type: 'estimate', value: 3 },
+        order: [
+          { term: DF.variable('p'), direction: 'asc' },
+          { term: DF.variable('g'), direction: 'asc' },
+        ],
+        canContainUndefs: false,
+        variables: [ DF.variable('p'), DF.variable('g') ],
+      });
+      await expect(output.bindingsStream).toEqualBindingsStream(
+        [
+          BF.bindings([
+            [ DF.variable('p'), DF.namedNode('p1') ],
+            [ DF.variable('g'), DF.namedNode('g1') ],
+          ]),
+          BF.bindings([
+            [ DF.variable('p'), DF.namedNode('p2') ],
+            [ DF.variable('g'), DF.namedNode('g2') ],
+          ]),
+          BF.bindings([
+            [ DF.variable('p'), DF.namedNode('p3') ],
+            [ DF.variable('g'), DF.namedNode('g3') ],
+          ]),
+        ],
+      );
+    });
+
+    it('should run <<?s p o>> ?p o ?g', async() => {
+      mediatorResolveQuadPattern.mediate = jest.fn(
+        () => {
+          const data = new ArrayIterator([
+            quad('<<s1 p o>>', 'p1', 'o1', 'g1'),
+            quad('<<s2 p o>>', 'p2', 'o2', 'g2'),
+            quad('<<s3 p o>>', 'p3', 'o3', 'g3'),
+            quad('sd', 'pd', 'od'),
+          ]);
+          data.setProperty('metadata', metadataContent);
+          return Promise.resolve({ data });
+        },
+      );
+
+      const operation = AF.createPattern(
+        DF.quad(DF.variable('s'), DF.namedNode('p'), DF.namedNode('o')),
+        DF.variable('p'),
+        DF.namedNode('o'),
```

Comment:
How does it happen that query matches quads where `o` is `[o1](https://github.com/comunica/comunica/pull/1222/commits/7acc31264c02321900b683850bc639738e0e3de6#diff-5b4a7cc2053820fe543450f4a971853e65a5e68a67855523613b6c352a997e1dR865)`, `o2`, `o3`?